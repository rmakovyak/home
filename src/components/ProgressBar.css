.ProgressBar {
  background-color: #efefef;
  border-radius: 0.375rem;
  height: 0.375rem;
  width: 100%;
  overflow: hidden;
  position: relative;
}

.ProgressBar__fill {
  background: #2e3642;
  border-radius: 0.375rem;
  height: 100%;
  transition: width 1s cubic-bezier(0.17, 0.67, 0.19, 0.95);
}

.ProgressBar__fill--0 {
  width: 1%;
}

.ProgressBar__fill--25 {
  width: 25%;
}

.ProgressBar__fill--50 {
  width: 50%;
}

.ProgressBar__fill--75 {
  width: 75%;
}

.ProgressBar__fill--100 {
  width: 100%;
}
