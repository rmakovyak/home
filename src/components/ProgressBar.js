import React from 'react';
import PropTypes from 'prop-types';

import styles from './ProgressBar.css';

function ProgressBar({ level }) {
  return (
    <div className="ProgressBar">
      <div className={`ProgressBar__fill ProgressBar__fill--${level}`} />
    </div>
  );
}

ProgressBar.propTypes = {
  level: PropTypes.oneOf([0, 1, 2, 3, 4])
};

ProgressBar.defaultProps = {
  level: 0
};

export default ProgressBar;
