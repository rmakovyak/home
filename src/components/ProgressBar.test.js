import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProgressBar from './ProgressBar';

configure({ adapter: new Adapter() });

test('render a ProgressBar', () => {
  const wrapper = shallow(<ProgressBar />);
  expect(wrapper).toMatchSnapshot();
});

test('render a ProgressBar with a level prop', () => {
  const wrapper = shallow(<ProgressBar level={2} />);
  expect(wrapper).toMatchSnapshot();
});
