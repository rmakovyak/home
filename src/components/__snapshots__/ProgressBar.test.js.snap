// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a ProgressBar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    level={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ProgressBar__fill ProgressBar__fill--0"
/>,
      "className": "ProgressBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "ProgressBar__fill ProgressBar__fill--0",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ProgressBar__fill ProgressBar__fill--0"
/>,
        "className": "ProgressBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ProgressBar__fill ProgressBar__fill--0",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`render a ProgressBar with a level prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    level={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ProgressBar__fill ProgressBar__fill--2"
/>,
      "className": "ProgressBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "ProgressBar__fill ProgressBar__fill--2",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ProgressBar__fill ProgressBar__fill--2"
/>,
        "className": "ProgressBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ProgressBar__fill ProgressBar__fill--2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
